### DISCUSSION TITLE
Need help to run a "java -jar" command

### DISCUSSION THREAD

**Discussion by Andysg1947** [gocd/gocd#12720] [Development]
Hi Team, 
I have a use case in which I want to run "java -jar {jarfile_from_another_repo} {current_repo_workspace} "
where jarfile_from_another_repo is a jar file stored in another repo and 
current_repo_workspace is the workspace of current repo in which we are running pipeline.

It would be helpful if you can help me to solve this use case.

Thanks

**Comment by sriramnrn** [comment]
On Sun, Apr 28, 2024 at 12:52 PM Andysg1947 ***@***.***>
wrote:

> Hi Team,
> I have a use case in which I want to run "java -jar
> {jarfile_from_another_repo} {current_repo_workspace} "
> where jarfile_from_another_repo is a jar file stored in another repo and
> current_repo_workspace is the workspace of current repo in which we are
> running pipeline.
>
> It would be helpful if you can help me to solve this use case.
>

Is that a maven repo or just a URL from which you would download the jar
file? Did you publish that jar to that other repo?

Assuming a basic set up of the jar being downloadable over http, you would
author a shell script that downloads the jar, places it into the current
directory, and then uses java to run the jar.

e.g.
#!/bin/bash
set -e
curl -o myapp.jar https://remoterepo/package/myapp-1.3.2.jar
java -jar myapp.jar

If you can share more, then we can help you tailor make your pipeline
design.

Here's how GoCD itself is built and tested using GoCD:
https://build.gocd.org/

Thanks

**Reply by Andysg1947** [reply]
Hi Sriram,

It is not maven repo, we have a jar file which will scan the code to generate a report, we have placed the jar file in a Repo called "scanner" and we have the main application in another repo called "main-app". 
Whenever we push any changes to "main-app" gocd pipeline will trigger and in the pipeline steps we want to fetch the jar file from scanner repo and store it in temporary folder which is accessible during the pipeline.
We have a command as "java -jar {path_of_jarfile_stored_in_temp_folder} {path_of_main-app_repo}"
Please let me know how to store the jar file from scanner repo to a temporary folder which is accessible in the above command.

**Reply by chadlwilson** [reply]
If these repos are both source control repos, e.g Git - if you add **both** repos as two different GoCD materials with different "alternate checkout paths" you should be able to have GoCD ensure both are available at deterministic paths.

So

**Material 1** - main-app - set alternate checkout path to `app`
**Material 2** - scanner - set alternate checkout path to `scanner`

Assuming task "working folder" is left blank, set task to do something like

`java -jar scanner/path/to/jar-file.jar app/`
